EIPS/eip-6997.md

```diff
@@ -0,0 +1,371 @@
+---
+eip: 6997
+title: ERC-721 with transaction validation step.
+description: A new validation step for transfer and approve calls, achieving a security step in case of stolen wallet.
+author: eduardfina (@eduardfina), Eduard López i Fina
+discussions-to: https://ethereum-magicians.org/t/erc721-with-a-validation-step/14071
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-07
+requires: 721
+---
+
+## Abstract
+
+This standard is an extension of [ERC-721](./eip-721.md). It defines a new validation functionality to avoid wallet draining, every `transfer` or `approve` will be locked waiting for validation.
+
+## Motivation
+
+The power of the Blockchain is at the same time its weakness, giving the user full responsibility for their data.
+
+Many cases of NFT theft currently exist, and current NFT anti-theft schemes, such as transferring NFTs to cold wallets, make NFTs inconvenient to be used.
+
+Having a validation step before every `transfer` and `approve` would bring the Smart Contract developers the possibility to create a security NFT anti-theft schemes.
+
+When I was developing this standard I was thinking of a system where there would be a validator address that would be responsible for validating all Smart Contract transactions.
+
+This address would be connected to a dApp where the user could see the validation requests of his NFTs and accept the correct ones.
+
+Giving this address only the power to validate transactions would make a much more secure system where to steal an NFT the thief would have to have both the user's address and the validator address simultaneously.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+[ERC-721](./eip-721.md) compliant contracts MAY implement this EIP.
+
+When `transfer` is called by the owner, the transfer MUST be frozen and a `TransferValidation` MUST be created.
+
+When `approve` is called by the owner, the transfer MUST be frozen and an `ApproveValidation` MUST be created.
```

Comment:
I'd recommend `ApprovalValidation` instead of `ApproveValidation` because `ApproveValidation` implies that "a validation was approved", instead of "an approval needs to be validated".

Reply:
Changed 👍